lib/src/ui/pages/settings/settings_page.dart

```diff
@@ -0,0 +1,225 @@
+                _buildDarkModeSection(context),
+                SizedBox(height: _Constants.sectionVerticalSpacing),
+                Divider(),
+                SizedBox(height: _Constants.sectionVerticalSpacing),
+                _buildPopularSection(context),
+                SizedBox(height: _Constants.sectionVerticalSpacing),
+                Divider(),
+                SizedBox(height: _Constants.sectionVerticalSpacing),
+                _buildAboutSection(context),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildAppBar(BuildContext context) => AppBar(
+        elevation: Dimens.appBarElevation,
+        iconTheme: IconThemeData(color: Colors.black54),
+        backgroundColor: Colors.white,
+      );
+
+  Widget _buildHeader(BuildContext context) => Text(
+        Strings.of(context).settings,
+        style: Theme.of(context).textTheme.headline3,
+      );
+
+  Widget _buildDarkModeSection(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.only(
+        left: _Constants.sectionLeftSpacing,
+        right: _Constants.sectionRightSpacing,
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildSettingsHeader(context, Strings.of(context).darkMode),
+          CupertinoSwitch(
+            value: _isDarkModeEnabled,
+            onChanged: (isSelected) {
+              setState(() {
+                _isDarkModeEnabled = isSelected;
+              });
+            },
+            activeColor: Colors.black54,
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildPopularSection(BuildContext context) {
+    return Padding(
+      padding: EdgeInsets.only(
+        left: _Constants.sectionLeftSpacing,
+        right: _Constants.sectionRightSpacing,
+      ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          _buildSettingsHeader(context, Strings.of(context).popular),
+          Column(
+            children: PopularNewsCriterion.values.map((criterion) {
+              return _buildPopularSectionItem(context, criterion);
+            }).toList(),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Padding _buildPopularSectionItem(BuildContext context, PopularNewsCriterion popularNewsCriterion) {
+    return Padding(
+      padding: const EdgeInsets.only(
+        left: _Constants.sectionItemLeftSpacing,
+        top: _Constants.sectionItemTopSpacing,
+        bottom: _Constants.sectionItemBottomSpacing,
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildPopularSettingText(context, popularNewsCriterion),
+          SizedBox(width: _Constants.switchLeftSpacing),
+          _buildSwitch(popularNewsCriterion == _selectedPopularNewsCriterion, (isSelected) {
+            if (_selectedPopularNewsCriterion != popularNewsCriterion && isSelected) {
```

Comment:
tu wystarczy samo `isSelected`, jak będziemy wyłączać obecnie zacheckowany check to `isSelected` będzie false i nie przejdzie dalej